File: src/main/java/com/example/creditreportservice/model/CreditReport.java
package com.example.creditreportservice.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

/**
 * Sample JPA entity representing a credit report record.
 *
 * <p>This is a starting template — adjust fields and constraints to match the
 * real domain model.
 */
@Entity
@Table(name = "credit_reports")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreditReport {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String subjectId;

    private Integer score;

    private String status;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;

}

File: internal/models/credit_report.go
package models

import "time"

// CreditReport is the row model for the credit_reports table.
type CreditReport struct {
	ID        int64      `json:"id"         db:"id"`
	SubjectID string     `json:"subjectId"  db:"subject_id"`
	Score     *int32     `json:"score"      db:"score"`
	Status    *string    `json:"status"     db:"status"`
	CreatedAt time.Time  `json:"createdAt"  db:"created_at"`
	UpdatedAt time.Time  `json:"updatedAt"  db:"updated_at"`
}
